content/guest-post/nextjs-authentication-guide/index.md

```diff
@@ -0,0 +1,428 @@
+
+Let’s update the `Nav.js` file with the code below to fix that:
+
+```javascript
+import { Button, Flex, Stack } from "@chakra-ui/react";
+import { useLRAuth } from "loginradius-react";
+
+export default function Nav() {
+	const { isAuthenticated, user, loginWithRedirect, logout } = useLRAuth();
+	return (
+		<Flex as='nav'>
+			<Stack spacing={[6, 8]}>
+				{!user && (
+					 <Button onClick={() => loginWithRedirect()}>
+						Login to continue
+					 </Button>
+				)}
+
+			 {isAuthenticated && user && (
+					<Button onClick={() => logout()}>
+						Log out
+				 </Button>
+			 )}
+		 </Stack>
+	 </Flex>
+ );
+}
+```
+We conditionally display the login and logout buttons based on the authentication state.
+
+If there is a logged-in user, we want to redirect them to the profile page. 
+
+Navigate to `pages/index.js` file and update it with the code below:
+
+```javascript
+import { useEffect } from "react";
+import { useRouter } from "next/router";
+import { useLRAuth } from "loginradius-react";
+import { Button, Heading, VStack, Center } from "@chakra-ui/react";
+
+export default function Home() {
+	const { isLoading, isAuthenticated, error, user } = useLRAuth();
+	const router = useRouter();
+
+	useEffect(() => {
+		if (user && isAuthenticated) {
+			router.push("/profile");
+		}
+	}, [router, user, isAuthenticated]);
+	
+
+	if (isLoading) {
+		return <div>Loading...</div>;
+	}
+
+	if (error) {
+		return <div>Oops... {error.message}</div>;
+	}
+
+	return (
+		<Center pt={10}>
+			<VStack spacing={[6, 8]}>
+				<Heading as="h2">Welcome</Heading>
+				<Text fontSize="3xl">Login to continue</Text>
+			</VStack>
+		</Center>
+	);
+}
+```
+
+Let’s break down the above code snippet:
+
+- In the `useEffect`, it checks if there is a user object and if `isAuthenticated` is true. If these conditions are true, it redirects the user to their profile page. These conditions will only be true when the user has logged in. You'll set up the profile page later.
+- It checks if the page is loading using the `isLoading` boolean. If true, it displays “loading…”. In a production-ready application, you would display a loading spinner.
+- If there is an error during the authentication process, it accesses the message from the `error` object and displays it to the user.
+* Finally, return some JSX.
+
+### Displaying Authenticated User Data
+
+You have to display information about the authenticated user. Let’s set that up.
+Create a profile.js file in the pages directory and paste the code below:
+```javascript
+import { Box, Center, Heading } from "@chakra-ui/react";
+import { useLRAuth } from "loginradius-react";
+
+export default function Profile() {
+	const { user, isLoading, isAuthenticated } = useLRAuth();
+	
+	if (isLoading) {
+		return <div>Loading...</div>;
+	}
+
+	if (isAuthenticated) {
+		return (
+			<Box>
+				<Nav />
```

Comment:
Returning `<Nav />` here is not necessary since we are returning `Nav` from the layout component for all pages.